From b7f5c807b80ac60f166b8da2d96fdb0eefe32a41 Mon Sep 17 00:00:00 2001
From: Sungtak Lee <taklee@google.com>
Date: Sun, 11 Dec 2022 06:16:15 +0000
Subject: [PATCH] C2SurfaceSyncObj: prevent OOB read in Import

Prevent OOB read in C2SurfaceSyncObj::Import from libcodec2_vndk.

Bug: 240140929
Test: Manual
Change-Id: I7b4cd8aa3fa5b9b2160f0eba40a618b4dd536d5c
(cherry picked from commit 9b4f38105ad66615e811483f4927942b231c84b7)
Merged-In: I7b4cd8aa3fa5b9b2160f0eba40a618b4dd536d5c
(cherry picked from commit e3958886dbdd65ac8020a4554c9e567f95a6d813)
Merged-In: I7b4cd8aa3fa5b9b2160f0eba40a618b4dd536d5c
---
 media/codec2/vndk/platform/C2SurfaceSyncObj.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/media/codec2/vndk/platform/C2SurfaceSyncObj.cpp b/media/codec2/vndk/platform/C2SurfaceSyncObj.cpp
index e55bdc0047..bbbd03efeb 100644
--- a/media/codec2/vndk/platform/C2SurfaceSyncObj.cpp
+++ b/media/codec2/vndk/platform/C2SurfaceSyncObj.cpp
@@ -64,6 +64,11 @@ std::shared_ptr<C2SurfaceSyncMemory> C2SurfaceSyncMemory::Import(
     }
 
     HandleSyncMem *o = static_cast<HandleSyncMem*>(handle);
+    if (o->size() < sizeof(C2SyncVariables)) {
+        android_errorWriteLog(0x534e4554, "240140929");
+        return nullptr;
+    }
+
     void *ptr = mmap(NULL, o->size(), PROT_READ | PROT_WRITE, MAP_SHARED, o->memFd(), 0);
 
     if (ptr == MAP_FAILED) {
-- 
2.39.1.456.gfc5497dd1b-goog

